plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx

```diff
@@ -23,19 +23,168 @@ import { Icon, close } from '@wordpress/icons';
+	variation,
+	iconSize,
+	style,
+	navigationStyle,
+}: {
+	variation: BlockVariationTriggerType;
+	iconSize: BlockAttributes[ 'iconSize' ];
+	style: BlockAttributes[ 'style' ];
+	navigationStyle: BlockAttributes[ 'navigationStyle' ];
+} ) => {
+	const overlayNavigationLabel = (
+		<OverlayNavigationLabel variation={ variation } />
+	);
+	const overlayNavigationIcon = (
+		<OverlayNavigationIcon
+			variation={ variation }
+			iconSize={ iconSize }
+			style={ style }
+		/>
+	);
+
+	if ( navigationStyle === 'label-and-icon' ) {
+		if ( variation === 'open-overlay' ) {
+			return (
+				<>
+					{ overlayNavigationIcon }
+					{ overlayNavigationLabel }
+				</>
+			);
+		} else if ( variation === 'close-overlay' ) {
+			return (
+				<>
+					{ overlayNavigationLabel }
+					{ overlayNavigationIcon }
+				</>
+			);
+		}
+	} else if ( navigationStyle === 'label-only' ) {
+		return overlayNavigationLabel;
+	} else if ( navigationStyle === 'icon-only' ) {
+		return overlayNavigationIcon;
+	}
+
+	return null;
+};
+
+type BlockProps = BlockEditProps< BlockAttributes > & { context: BlockContext };
+
+export const Edit = ( { attributes, setAttributes, context }: BlockProps ) => {
+	const { navigationStyle, buttonStyle, iconSize, style, triggerType } =
+		attributes;
+	const { 'woocommerce/product-filters/overlay': productFiltersOverlayMode } =
+		context;
 	const blockProps = useBlockProps( {
 		className: clsx( 'wc-block-product-filters-overlay-navigation', {
 			'wp-block-button__link wp-element-button': buttonStyle !== 'link',
 		} ),
 	} );
+	const {
+		isWithinProductFiltersTemplatePart,
+		isWithinProductFiltersOverlayTemplatePart,
+	}: {
+		isWithinProductFiltersTemplatePart: boolean;
+		isWithinProductFiltersOverlayTemplatePart: boolean;
+	} = useSelect( ( select ) => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		const { getCurrentPostId, getCurrentPostType } =
+			select( 'core/editor' );
+		const currentPostId = getCurrentPostId< string >();
+		const currentPostIdParts = currentPostId?.split( '//' );
+		const currentPostType = getCurrentPostType< string >();
+		let isProductFiltersTemplatePart = false;
+		let isProductFiltersOverlayTemplatePart = false;
+
+		if (
+			currentPostType === 'wp_template_part' &&
+			currentPostIdParts?.length > 1
+		) {
+			const [ , postId ] = currentPostIdParts;
+			isProductFiltersTemplatePart = postId === 'product-filters';
+			isProductFiltersOverlayTemplatePart =
+				postId === 'product-filters-overlay';
+		}
+
+		return {
+			isWithinProductFiltersTemplatePart: isProductFiltersTemplatePart,
+			isWithinProductFiltersOverlayTemplatePart:
+				isProductFiltersOverlayTemplatePart,
+		};
+	} );
+
+	const shouldHideBlock = () => {
+		if ( triggerType === 'open-overlay' ) {
+			if (
+				productFiltersOverlayMode ===
+				ProductFiltersBlockOverlayAttribute.NEVER
+			) {
+				return true;
+			}
+
+			if ( isWithinProductFiltersTemplatePart ) {
+				return true;
+			}
```

Comment:
This always hides the navigation block when editing the Product Filters template part. Removing it fixes the issue.

Reply:
Nice catch! I fixed it in 9d2a73e9bc8511c3be7912c393097cfc2416e016